### ISSUE TITLE
Attach street-level imagery tags to changesets

### ISSUE THREAD

**Issue by cbeddow** [openstreetmap/iD#6279]
When a user opens any of the street-level imagery layers:

* Bing Streetside
* Mapillary
* OpenStreetCam

I suggest that the changeset automatically adds:

* source=mapillary
* source= streetlevel imagery
* hashtag=#mapillary

![image](https://user-images.githubusercontent.com/10554922/57023225-5e1e1700-6c29-11e9-982f-4e56fc8a71dd.png)

This can be cleared by the user if they don't want to add it. This is helpful for tracking generally how many changesets use street-level imagery, and then knowing the total proportion that use each provider. For Mapillary this helps us to know where out imagery is most useful (and potentially give image capturing users feedback on this so they know they are helping contribute to OSM even when another user is using the images)

**Comment by quincylvania** [comment]
This could be useful! To reduce false positives, perhaps we could only add the values if a user edits a feature near the photo location with the photo open.

Thinking ahead, if we make iD really good at this then we could possible update the `source` tag on the feature itself and _maybe_ do automatic photo-linking (#6196). Though I'd still prefer a manual photo-linking UI first.

**Comment by pnorman** [comment]
👎 to a hashtag, it's unnecessary with a source tag.

**Comment by cbeddow** [comment]
Just one or other makes sense then, yes. Source is actually useful, hashtag just visible.

The photo linking would be great but too complex for now, yeah. We could even start by sequence-linking (each photo is part of a collection of photos that are a sequence, so it helps).

**Comment by cbeddow** [comment]
Also, I am not sure what radius would make sense on having the distance check. 200 meters maybe? 

Additionally, would it make sense to have `data overlay` as a source as well, for example with Mapillary traffic signs layer? So it would have `source=mapillary` and `source=data overlay` so that it's differentiated from `streetlevel imagery`, although it could be both, if both are active and used. Can reconsider this at another time if it complicates things though.

**Comment by quincylvania** [comment]
@cbeddow FYI it looks like iD already includes info on the photo layers in the `imagery_used` changeset tag if the user makes an edit with an overlay enabled. Perhaps this is enough to get you started? For example:

```
imagery_used=DigitalGlobe Premium Imagery;Bing Streetside;Mapillary Images;OpenStreetCam Images
```

There are a number of reasons this should be improved though:

- If the user opens a lot of imagery then these values might not fit in the tag, which is limited to 255 characters.
- The user can't edit it specify whether they actually used these as sources.
- There's plenty more information that iD could pass on, like the sequence or photos actually opened during edits.

**Comment by quincylvania** [comment]
@cbeddow I made it so the relevant photo overlay values are also added to the `source` tag, where users can edit them if needed.

**Comment by tordans** [comment]
@quincylvania I think there is a bug with this new feature. But my test case is not great.

1. I opened http://preview.ideditor.com/master/#background=Berlin-2018&disable_features=boundaries&locale=en&map=18.00/52.47342/13.44771 via bookmark
1. I delete something, then added Mapillary, clicked around, deleted more …
1. I checked the source-filed on the save-screen which was first empty but later had data <--- bad test case :-)
1. I tried to reproduce my test case, closed the map, re-opened
1. Choose "delete all data from the last session" on the starting prompt
1. Did not open any Mapillary layer
1. Changed something random
1. => The save-screen sho still shows sources, even though I never added them to this session
I expected the source area to only check what I did in this session; or maybe even only in this commit-cycle (so it would reset after I saved) and re-check what sources I used before saving again …

Might this be related to the commit-tagging-feature, which has also a memory feature between session?

<img width="766" alt="Bildschirmfoto 2019-05-18 um 07 43 47" src="https://user-images.githubusercontent.com/111561/57965175-e64a2f00-7940-11e9-83c4-2b2199c8b5dc.png">

--- 

(Unrelated to this and more for myself: A ping from here to https://github.com/openstreetmap/iD/issues/6300.)

**Comment by quincylvania** [comment]
@tordans Interesting. This is definitely because some of the changeset tags are persisted in local storage. Not sure if this is a bug or a feature.

**Comment by quincylvania** [comment]
@tordans I just pushed a fix for #6642 that should solve the bug you were seeing. Previously, the changeset comment, sources, and hashtags would persist even after saving or discarding your changes. Now these are cleared whenever the edit history is cleared.
